package/kedro_viz/integrations/kedro/hooks_utils.py

```diff
@@ -0,0 +1,110 @@
+"""Utility functions for Kedro hooks implementation."""
+
+import json
+import logging
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import fsspec
+from kedro.pipeline.node import Node as KedroNode
+
+from kedro_viz.constants import VIZ_METADATA_ARGS
+from kedro_viz.launchers.utils import _find_kedro_project
+from kedro_viz.utils import _hash, _hash_input_output
+
+logger = logging.getLogger(__name__)
+
+
+def hash_node(node: Any) -> str:
+    """Stable ID for KedroNode or I/O reference."""
+    return _hash(str(node)) if isinstance(node, KedroNode) else _hash_input_output(node)
+
+
+def make_dataset_event(
+    when: str,
```

Comment:
I don't think we should call this when, i think we can better name -- it's confusing when reading code